### ISSUE TITLE
Custom iOS simulator shell script

### ISSUE THREAD

**Issue by kastiglione** [bazelbuild/rules_apple#225]
Is it possible to provide a custom local alternative to [`ios_sim.template.sh`](https://github.com/bazelbuild/rules_apple/blob/master/apple/bundling/runners/ios_sim.template.sh)? I see that the default comes from the [`_runner_template`](https://github.com/bazelbuild/rules_apple/blob/658dd7b5423ee35a134784b31bea05f7a1b37030/apple/bundling/rule_factory.bzl#L196-L201) attribute, but I'm not sure how to pass this from the top level rules.

**Comment by sergiocampama** [comment]
No, it's not possible. The `ios_sim_runner` rule is not meant to be configurable that way. If you want to have another way of running tests, the recommended way is to create a new runner rule similar to `ios_sim_runner` and provide that one as the `runner` attribute in the test targets.

**Comment by kastiglione** [comment]
> the recommended way is to create a new runner rule similar to `ios_sim_runner` and provide that one as the `runner` attribute in the test targets

This appears to only be a solution to running tests, but not for running apps in the simulator.

**Comment by sergiocampama** [comment]
Oh, my mistake, I associated the runner attribute to the test targets. Are you looking for a missing feature in particular? Could this be added to the `ios_sim.template.sh`?

A workaround that you can use is create a script that builds the `ios_application` target, and then processes the output with another script that launches the simulator and runs the app

**Comment by kastiglione** [comment]
Didn't mean to close. I'll follow up to your questions shortly.

**Comment by keith** [comment]
> Are you looking for a missing feature in particular? Could this be added to the `ios_sim.template.sh`?

Not really, really we just want to break some of the hermiticity guarantees in order to have a nicer build, run, debug cycle. For most cases if you did a `bazel run` wiping the entire simulator wouldn't be a feasible way to work on an app. This is of course assuming you're using this for some reason over building in Xcode.

**Comment by sergiocampama** [comment]
Is that something that could be added to the runner that we already have? I don't want to increase the API surface since the runner entrypoints would need to be standardized.

**Comment by rahul-malik** [comment]
@sergiocampama @Keith - Could we just add a `--define` to avoid cleaning up the simulator if that's the only change in the script?
